tests/utils/test_utilities.py

```diff
@@ -0,0 +1,29 @@
+from xgi.utils import XGICounter, get_dual, load_xgi_data
+
+
+def test_load_xgi_data():
+    H = load_xgi_data("email-enron")
+    assert H.num_nodes == 148
+    assert H.num_edges == 10885
+    assert H["name"] == "email-Enron"
+    assert H.nodes["4"]["name"] == "robert.badeer@enron.com"
+    assert H.edges["0"]["timestamp"] == "2000-01-11T10:29:00"
+
+
+def test_get_dual(dict5):
+    dual = get_dual(dict5)
+    assert dual[0] == [0]
+    assert dual[1] == [0]
+    assert dual[2] == [0]
+    assert dual[3] == [0]
+    assert dual[4] == [1]
+    assert dual[5] == [2]
+    assert dual[6] == [2, 3]
+    assert dual[7] == [3]
+    assert dual[8] == [3]
+
+
+def test_xgi_counter():
```

Comment:
Can we replace XGICounter with python's [count](https://docs.python.org/3/library/itertools.html#itertools.count)? If so, we can leave that for a separate PR but we should as much as possible make use of what's already in the standard library.

Reply:
I agree! I'm not familiar with `count` until now, but I will make an issue to deal with later.

Reply:
Actually, I'm not sure that it does the same thing. `XGICounter` increments a counter every time that it is called, which doesn't seem to be the same as `count` from what I could tell.